cats-effect/src/main/scala/com/avast/sst/catseffect/syntax/TimeSyntax.scala

```diff
@@ -0,0 +1,38 @@
+package com.avast.sst.catseffect.syntax
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect.syntax.bracket._
+import cats.effect.{Bracket, Clock}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.avast.sst.catseffect.syntax.TimeSyntax.FOps
+
+import scala.concurrent.duration.Duration
+
+trait TimeSyntax {
+
+  implicit def sstFOps[F[_], A](f: F[A]): FOps[F, A] = new FOps(f)
+
+}
+
+object TimeSyntax {
+
+  final class FOps[F[_], A](private val f: F[A]) extends AnyVal {
+
+    /** Measures the time it takes the effect to finish and records it using the provided function. */
+    def time(record: Duration => F[Unit])(implicit F: Bracket[F, Throwable], C: Clock[F]): F[A] = {
+      val unit = TimeUnit.NANOSECONDS
```

Comment:
What is this for? Why don't you either a) use it directly b) have it as `val` in the `object`?

Reply:
Field definitions are not allowed in value classes. And the only reason to have the variable is to not make a mistake in two places where it's used.